### ISSUE TITLE
(JavaScript) Incorrect highlighting for zero-width block comments

### ISSUE THREAD

**Issue by esqew** [highlightjs/highlight.js#3014]
**Describe the issue**
It appears that syntactically-valid zero-width JavaScript block comments (`/**/`) are not being highlighted correctly. The [ECMAScript standard](https://262.ecma-international.org/11.0/#prod-MultiLineComment) specifically marks characters between `/*` and `*/` tokens as optional, so I'd expect `/**/` to be highlighted as a standalone comment, instead of allowing the comment style highlighting to bleed over until the next occurrence of a `*/` sequence.

**Which language seems to have the issue?**
JavaScript

**Are you using `highlight` or `highlightAuto`?**
Issue appears to manifest in both modes.

**Sample Code to Reproduce**
```
console.log("first line") /**/
console.log("second line") /**/
console.log("third line")
```

![Screenshot demonstrating incorrect syntax highlighting](https://user-images.githubusercontent.com/836198/108106969-9ee45b80-705c-11eb-90d7-b00c268d5229.png)

[JSFiddle reproduction](https://jsfiddle.net/nwq83k0h/)

**Expected behavior**
Comment is highlighted correctly and the appropriate highlighting doesn't bleed until the next occurrence of `*/`, similar to how the syntax highlighter employed on JSFiddle handles it (screenshot below):

![Demonsration of correct syntax highlighting](https://user-images.githubusercontent.com/836198/108106602-2a112180-705c-11eb-9cf4-ab1e3af2f77c.png)

**Additional context**
I discovered this while browsing Stack Overflow, specifically on [this question](https://stackoverflow.com/questions/66229659/is-there-a-way-to-get-the-index-of-the-filtered-array-while-filtering).

It seems to me that this case should hit [`C_BLOCK_COMMENT_MODE`](https://github.com/highlightjs/highlight.js/blob/60734e71fd1f6c24fabf050a95d00420e3f67bf6/src/lib/modes.js#L85), which is declared as [a `variant` within `javascript.js`' comment definition](https://github.com/highlightjs/highlight.js/blob/eb2a0305957b0bd47cdf84df7cabd7327851b74f/src/languages/javascript.js#L174).

**Comment by joshgoebel** [comment]
Ah, this matches the `/**` rule... we need a special case just for `/**/`.

**Comment by esqew** [comment]
My apologies - this seems to have been fixed previously in #2896 / #2897. Any symptoms of this issue that remain would be due to running a version of highlight.js < 10.4.1.
